spyder/plugins/editor/widgets/editor.py

```diff
@@ -2329,6 +2329,14 @@ def __refresh_readonly(self, index):
             if not osp.isfile(finfo.filename):
                 # This is an 'untitledX.py' file (newly created)
                 read_only = False
+            try:
+                # Try to open the file to see if the permissions allow writing
+                # in Windows. For further info, see issue
+                # https://github.com/spyder-ide/spyder/issues/10657
+                fd = open(finfo.filename, os.O_RDWR)
+                fd.close()
+            except PermissionError:

```

Comment:
`PermissionError` doesn't exist in Python 2. I think in Python 3 it inherits from `OSError`, so that's the error we need to capture here.